internal/debug/debug.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug // import "go.opentelemetry.io/otel/internal/debug"
+
+import (
+	"log"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/stdr"
+)
+
+// globalLogger is the logging interface used within the otel api and sdk provide deatails of the internals.
+//
+// The default logger uses stdr which is backed by the standard `log.Logger`
+// interface. This logger will only show messages at the Error Level.
+var globalLoggger logr.Logger = stdr.New(log.New(os.Stderr, "", log.LstdFlags|log.Lshortfile))
+
+// SetLogger overrides the globalLogger with l.
+//
+// To see Info messages use a logger with `l.V(1).Enabled() == true`
+// To see Debug messages use a logger with `l.V(5).Enabled() == true`
+func SetLogger(l logr.Logger) {
+	globalLoggger = l
+}
+
+// Info prints messages about the general state of the API or SDK.
+// This should usually be less then 5 messages a minute
+func Info(msg string, keysAndValues ...interface{}) {
+	globalLoggger.V(1).Info(msg, keysAndValues...)
+}
+
+// Error prints messages about exceptional states of the API or SDK.
+func Error(err error, msg string, keysAndValues ...interface{}) {
```

Comment:
Do we intend to log transient failures? If so then on which level? I tend to use `Warn` for such cases

Reply:
So I think it would depend.  My general idea with this is if someone were to turn on V(1), they would get something like 5-10 messages a minute or so.  This number was picked from thin air to represent a low but manageable number of messages.  Things that describe the current state of the SDK.  
If your transient failure was something that needed immediate attention then it would be in Error, which all users would see.  If not then it would be in Info, which describes the last known state of the SDK, and some attempt should be made to no have:
```
10:01:04.05  Failed to connect
10:01:04.06  Failed to connect
10:01:04.07  Failed to connect
10:01:04.08  Failed to connect
10:01:04.09  Failed to connect
```

Reply:
👍  resolving